### ISSUE TITLE
[HOLD App#50650] Tracking: Remove VisionCamera `CameraSession on Fabric`patch once the library was fully migrated to New Arch

### ISSUE THREAD

**Issue by chrispader** [Expensify/App#49988]
### Description

There is a bug in `react-native-vision-camera` where the `cameraSession` isn't properly deinitialized/recycled on Fabric under RN New Architecture.

Since we currently already have a patch file to make RNVC compatible with React Native `v0.75`, we cannot directly apply this change upstream. We should also wait to fix this until the library has fully been migrated to the New Architecture.

cc @mountiny 

<details><summary>Issue Owner</summary>Current Issue Owner: @chrispader</details>

**Comment by mountiny** [comment]
I believe we can close this one as the payment will be handled as part of the native stack pr

**Comment by chrispader** [comment]
@mountiny didn't we create this issue to track whether this got fixed upstream? This is not yet fixed in VIsionCamera

**Comment by mountiny** [comment]
Whoops you are right, thank you for the ping, I skimmed over this and havent realized.

**Comment by mountiny** [comment]
@chrispader how is this looking?

**Comment by chrispader** [comment]
I will try to add this to `react-native-vision-camera` this week! Shouldn't be a big change

**Comment by mountiny** [comment]
@hannojg just to confirm, have you tried working on something similar and it showed to be a lot of work?

**Comment by hannojg** [comment]
@mountiny yes, will discuss with Chris internally first and then get back to you! Thanks for bringing this up

**Comment by chrispader** [comment]
@mountiny so i just had a quick call with @hannojg about this change and our path forward with VisionCamera.

Hanno already explained in [this thread](https://margelo.slack.com/archives/C035J5C9FAP/p1730738516530279) why we won't migrate `react-native-vision-camera` to RN New Arch. Essentially we will be migrating the library to Nitro modules soon and the changes needed for new arch support are just too complex and unnecessary.

Therefore, we also won't be able to add the changes from the patch discussed in this PR to the library. This is because of two reasons:

- This patch (`cameraSession` fix under New Arch) is based on another patch that we have in E/App, which adds/migrates some of the VisionCamera legacy view components to Fabric components
- In the upstream library we don't have any Fabric related code in VisionCamera atm, so we also cannot add the changes from this patch (`prepareForRecycle`)

**Comment by chrispader** [comment]
Since we are in the process of migrating the library to Nitro modules, it would be beneficial to update to a Nitro-enabled version soon.

@hannojg created an issue for tracking the update of `react-native-vision-camera` to a newer version in https://github.com/Expensify/App/issues/50650

We can also keep track of the `CameraSession` fix over there and make sure this works with Nitro. We should therefore keep the current patch as-is and remove it once we migrated to Nitro.

@mountiny does that sound good to you?

**Comment by mountiny** [comment]
Sounds great, thanks!

**Comment by mountiny** [comment]
On hold

**Comment by chrispader** [comment]
No update on the VisionCamera -> Nitro Views migration yet. Jfyi, Nitro Views will require at least RN 0.78, so it isn't really an option yet anyway.

**Comment by mountiny** [comment]
@chrispader What is the latest here?

**Comment by chrispader** [comment]
The migration is already in the working, but this currently has a low priority. A blocker at the moment is that it would require a minimum RN version of 0.78 from all `react-native-vision-camera` users and we're not sure yet whether we can pre-requisite that from everyone already.

I think we're going to have to keep this open for a bit longer still, the migration ⏳

**Comment by mountiny** [comment]
Not sure about this one anymore vut could we apply a patch?

**Comment by chrispader** [comment]
> Not sure about this one anymore vut could we apply a patch?

@mountiny The reason for this tracking issue was to remove the patch we currently have for VisionCamera once it has been migrated to Nitro Views. This will still take some time, so either we leave it open or we disregard this for the moment

**Comment by mountiny** [comment]
Ok great, sorry lots of things happening haha. Can you please update the patch info following this format? https://github.com/Expensify/App/pull/61498

Thanks!

**Comment by chrispader** [comment]
> Ok great, sorry lots of things happening haha. Can you please update the patch info following this format? [#61498](https://github.com/Expensify/App/pull/61498)
> 
> Thanks!

@mountiny will do!

**Comment by chrispader** [comment]
@mountiny created a PR that adds the patch info and documentation here: https://github.com/Expensify/App/pull/63966

**Comment by mountiny** [comment]
Keeping this monthly

**Comment by chrispader** [comment]
@mountiny as we've removed all patches for VisionCamera in [this PR](https://github.com/Expensify/App/pull/66640), this can actually be kept closed

**Comment by mountiny** [comment]
Ok thanks for following up
